packages/@uppy/tus/src/index.ts

```diff
@@ -280,7 +278,8 @@ export default class Tus<M extends Meta, B extends Body> extends BasePlugin<
           // This means we can hold the Tus retry here with a `Promise.all`,
           // together with the returned value of the user provided
           // `onBeforeRequest` option callback (in case it returns a promise).
-          return Promise.all([p, userProvidedPromise])
+          await Promise.all([p, userProvidedPromise])
+          return undefined
```

Comment:
are we sure about returning something else here?

Reply:
not 100% sure, but the type of onBeforeRequest expects `Promise<void>` to be returned, and all other code paths return void, and I think it would be weird if someone expects an array with two elements of undefined as with the current code, e.g. return type `Promise<[void, void]>`